File: Assets/Share/Scripts/Communicate/Encrypt/AesEncryptionHelper.cs
namespace Communicate.Encrypt {
    public class AesEncryptionHelper {
        public AesEncryptionHelper(AesEncryption encryption, IObfuscate obfuscate) {
            _encryption = encryption;
            _obfuscate = obfuscate;
        }

        private readonly AesEncryption _encryption;
        private readonly IObfuscate _obfuscate;

        public string Encrypt(string plainText) {
            var newAes = new AesEncryption();
            newAes.GenerateKey();
            var iv = newAes.GetIv();
            newAes.SetKeyAndIv(_encryption.GetKey(), iv);

            var encrypted = newAes.Encrypt(plainText);
            var mergedBytes = ByteUtils.MergeByteArray(
                ByteUtils.Base64ToByteArray(iv), ByteUtils.Base64ToByteArray(encrypted));
            return _obfuscate.Obfuscate2(mergedBytes);
        }

        public string Decrypt(string encryptedData) {
            var bytes = _obfuscate.DeObfuscate2(encryptedData);
            var iv = ByteUtils.ByteArrayToBase64(bytes[.._encryption.IvSize]);
            var cipherText = ByteUtils.ByteArrayToBase64(bytes[_encryption.IvSize..]);

            var newAes = new AesEncryption();
            newAes.SetKeyAndIv(_encryption.GetKey(), iv);

            var decrypted = newAes.Decrypt(cipherText);
            return decrypted;
        }

        public string Decrypt(byte[] encryptedData) {
            var bytes = _obfuscate.DeObfuscateBytes(encryptedData);
            var iv = ByteUtils.ByteArrayToBase64(bytes[.._encryption.IvSize]);
            var cipherText = ByteUtils.ByteArrayToBase64(bytes[_encryption.IvSize..]);

            var newAes = new AesEncryption();
            newAes.SetKeyAndIv(_encryption.GetKey(), iv);

            var decrypted = newAes.Decrypt(cipherText);
            return decrypted;
        }
    }
}

File: unity-telegram-template/src/encrypt/AesEncryptionHelper.ts
import IObfuscate from "./IObfuscate.ts";
import AesEncryption from "./AesEncryption.ts";
import {base64ToByteArray, byteArrayToBase64, mergeByteArray} from "../utils/String.ts";

export default class AesEncryptionHelper {
    constructor(
        private readonly _aes: AesEncryption,
        private readonly _obfuscate: IObfuscate,
    ) {
    }

    encrypt(plainText: string): string {
        const iv = this._aes.generateIV();
        const encrypted = this._aes.encrypt(plainText, iv);
        const mergedBytes = mergeByteArray(base64ToByteArray(iv), (base64ToByteArray(encrypted)));
        return this._obfuscate.obfuscate2(mergedBytes);
    }

    decrypt(encryptedData: string): string | null {
        const bytes = this._obfuscate.deobfuscate2(encryptedData);
        const iv = byteArrayToBase64(bytes.slice(0, this._aes.ivSize));
        const data = byteArrayToBase64(bytes.slice(this._aes.ivSize));
        return this._aes.decrypt(data, iv);
    }
}